src/notify_keys.rs

```diff
@@ -0,0 +1,53 @@
+use {
+    crate::websocket_service::decode_key,
+    rand_chacha::{
+        rand_core::{RngCore, SeedableRng},
+        ChaCha20Rng,
+    },
+    relay_rpc::domain::Topic,
+    url::Url,
+};
+
+pub struct NotifyKeys {
+    pub domain: String,
+    pub key_agreement_secret: x25519_dalek::StaticSecret,
+    pub key_agreement_public: x25519_dalek::PublicKey,
+    pub key_agreement_topic: Topic,
+    pub authentication_secret: ed25519_dalek::SigningKey,
+    pub authentication_public: ed25519_dalek::VerifyingKey,
+}
+
+impl NotifyKeys {
+    pub fn new(notify_url: &str, keypair_seed: &str) -> Self {
```

Comment:
Why not return a `Result<Self, FromHexError>` instead of `Self` so we can actually remove the unwrap now?

Reply:
I'll do that, didn't know the wrapping function would handle the conversion